// @coreui/angular/esm2020/lib/nav/nav.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { NavComponent } from './nav.component';
import { NavItemComponent } from './nav-item.component';
import { NavLinkDirective } from './nav-link.directive';
import * as i0 from "@angular/core";
export class NavModule {
}
NavModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: NavModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NavModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: NavModule, declarations: [NavComponent, NavItemComponent, NavLinkDirective], imports: [CommonModule], exports: [NavComponent, NavItemComponent, NavLinkDirective] });
NavModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: NavModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: NavModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [NavComponent, NavItemComponent, NavLinkDirective],
                    imports: [
                        CommonModule
                    ],
                    exports: [NavComponent, NavItemComponent, NavLinkDirective]
                }]
        }] });
