// @coreui/angular/esm2020/lib/card/card.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CardComponent } from './card.component';
import { CardBodyComponent } from './card-body.component';
import { CardFooterComponent } from './card-footer.component';
import { CardHeaderComponent } from './card-header.component';
import { CardHeaderActionsComponent } from './card-header-actions.component';
import { CardGroupComponent } from './card-group.component';
import { CardImgDirective } from './card-img.directive';
import { CardLinkDirective } from './card-link.directive';
import { CardSubtitleDirective } from './card-subtitle.directive';
import { CardTextDirective } from './card-text.directive';
import { CardTitleDirective } from './card-title.directive';
import { CardImgOverlayComponent } from './card-img-overlay/card-img-overlay.component';
import * as i0 from "@angular/core";
export class CardModule {
}
CardModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: CardModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CardModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: CardModule, declarations: [CardComponent,
        CardBodyComponent,
        CardFooterComponent,
        CardGroupComponent,
        CardHeaderComponent,
        CardHeaderActionsComponent,
        CardImgDirective,
        CardImgOverlayComponent,
        CardLinkDirective,
        CardSubtitleDirective,
        CardTextDirective,
        CardTitleDirective], imports: [CommonModule], exports: [CardComponent,
        CardBodyComponent,
        CardFooterComponent,
        CardGroupComponent,
        CardHeaderComponent,
        CardHeaderActionsComponent,
        CardImgDirective,
        CardImgOverlayComponent,
        CardLinkDirective,
        CardSubtitleDirective,
        CardTextDirective,
        CardTitleDirective] });
CardModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: CardModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: CardModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        CardComponent,
                        CardBodyComponent,
                        CardFooterComponent,
                        CardGroupComponent,
                        CardHeaderComponent,
                        CardHeaderActionsComponent,
                        CardImgDirective,
                        CardImgOverlayComponent,
                        CardLinkDirective,
                        CardSubtitleDirective,
                        CardTextDirective,
                        CardTitleDirective,
                    ],
                    imports: [
                        CommonModule,
                    ],
                    exports: [
                        CardComponent,
                        CardBodyComponent,
                        CardFooterComponent,
                        CardGroupComponent,
                        CardHeaderComponent,
                        CardHeaderActionsComponent,
                        CardImgDirective,
                        CardImgOverlayComponent,
                        CardLinkDirective,
                        CardSubtitleDirective,
                        CardTextDirective,
                        CardTitleDirective,
                    ]
                }]
        }] });
