// @coreui/angular/esm2020/lib/header/header.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HeaderComponent } from './header/header.component';
import { HeaderBrandComponent } from './header-brand/header-brand.component';
import { HeaderDividerComponent } from './header-divider/header-divider.component';
import { HeaderNavComponent } from './header-nav/header-nav.component';
import { HeaderTextComponent } from './header-text/header-text.component';
import { HeaderTogglerDirective } from './header-toggler/header-toggler.directive';
import * as i0 from "@angular/core";
export class HeaderModule {
}
HeaderModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: HeaderModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
HeaderModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: HeaderModule, declarations: [HeaderComponent,
        HeaderBrandComponent,
        HeaderDividerComponent,
        HeaderNavComponent,
        HeaderTextComponent,
        HeaderTogglerDirective], imports: [CommonModule], exports: [HeaderComponent,
        HeaderBrandComponent,
        HeaderDividerComponent,
        HeaderNavComponent,
        HeaderTextComponent,
        HeaderTogglerDirective] });
HeaderModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: HeaderModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: HeaderModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        HeaderComponent,
                        HeaderBrandComponent,
                        HeaderDividerComponent,
                        HeaderNavComponent,
                        HeaderTextComponent,
                        HeaderTogglerDirective
                    ],
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        HeaderComponent,
                        HeaderBrandComponent,
                        HeaderDividerComponent,
                        HeaderNavComponent,
                        HeaderTextComponent,
                        HeaderTogglerDirective
                    ]
                }]
        }] });
