// @coreui/angular/esm2020/lib/table/table.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { TableDirective } from './table.directive';
import { TableColorDirective } from './table-color.directive';
import { TableActiveDirective } from './table-active.directive';
import * as i0 from "@angular/core";
export class TableModule {
}
TableModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: TableModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
TableModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: TableModule, declarations: [TableDirective,
        TableColorDirective,
        TableActiveDirective], imports: [CommonModule], exports: [TableDirective,
        TableColorDirective,
        TableActiveDirective] });
TableModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: TableModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: TableModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        TableDirective,
                        TableColorDirective,
                        TableActiveDirective
                    ],
                    exports: [
                        TableDirective,
                        TableColorDirective,
                        TableActiveDirective
                    ],
                    imports: [
                        CommonModule
                    ]
                }]
        }] });
