// @coreui/angular/esm2020/lib/button/button.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ButtonDirective } from './button.directive';
import { ButtonCloseDirective } from './button-close.directive';
import * as i0 from "@angular/core";
export class ButtonModule {
}
ButtonModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: ButtonModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ButtonModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: ButtonModule, declarations: [ButtonDirective,
        ButtonCloseDirective], imports: [CommonModule], exports: [ButtonDirective,
        ButtonCloseDirective] });
ButtonModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: ButtonModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: ButtonModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        ButtonDirective,
                        ButtonCloseDirective,
                    ],
                    imports: [
                        CommonModule,
                    ],
                    exports: [
                        ButtonDirective,
                        ButtonCloseDirective,
                    ]
                }]
        }] });
