// @coreui/angular/esm2020/lib/navbar/navbar.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { NavbarComponent } from './navbar.component';
import { NavbarBrandDirective } from './navbar-brand/navbar-brand.directive';
import { NavbarNavComponent } from './navbar-nav/navbar-nav.component';
import { NavbarTextComponent } from './navbar-text/navbar-text.component';
import { NavbarTogglerDirective } from './navbar-toggler/navbar-toggler.directive';
import * as i0 from "@angular/core";
export class NavbarModule {
}
NavbarModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: NavbarModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
NavbarModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: NavbarModule, declarations: [NavbarComponent,
        NavbarNavComponent,
        NavbarTextComponent,
        NavbarBrandDirective,
        NavbarTogglerDirective], imports: [CommonModule,
        RouterModule], exports: [NavbarComponent,
        NavbarNavComponent,
        NavbarTextComponent,
        NavbarBrandDirective,
        NavbarTogglerDirective] });
NavbarModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: NavbarModule, imports: [CommonModule,
        RouterModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: NavbarModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        NavbarComponent,
                        NavbarNavComponent,
                        NavbarTextComponent,
                        NavbarBrandDirective,
                        NavbarTogglerDirective
                    ],
                    exports: [
                        NavbarComponent,
                        NavbarNavComponent,
                        NavbarTextComponent,
                        NavbarBrandDirective,
                        NavbarTogglerDirective
                    ],
                    imports: [
                        CommonModule,
                        RouterModule
                    ]
                }]
        }] });
