// @coreui/angular/esm2020/lib/callout/callout.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CalloutComponent } from './callout.component';
import * as i0 from "@angular/core";
export class CalloutModule {
}
CalloutModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: CalloutModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
CalloutModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: CalloutModule, declarations: [CalloutComponent], imports: [CommonModule], exports: [CalloutComponent] });
CalloutModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: CalloutModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: CalloutModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [CalloutComponent],
                    exports: [CalloutComponent],
                    imports: [
                        CommonModule
                    ]
                }]
        }] });
