// @coreui/angular/esm2020/lib/widget/widget.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { CardModule } from '../card';
import { GridModule } from '../grid';
import { WidgetStatAComponent } from './widget-stat-a/widget-stat-a.component';
import { WidgetStatBComponent } from './widget-stat-b/widget-stat-b.component';
import { WidgetStatCComponent } from './widget-stat-c/widget-stat-c.component';
import { WidgetStatDComponent } from './widget-stat-d/widget-stat-d.component';
import { WidgetStatEComponent } from './widget-stat-e/widget-stat-e.component';
import { WidgetStatFComponent } from './widget-stat-f/widget-stat-f.component';
import * as i0 from "@angular/core";
export class WidgetModule {
}
WidgetModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: WidgetModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
WidgetModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: WidgetModule, declarations: [WidgetStatAComponent,
        WidgetStatBComponent,
        WidgetStatCComponent,
        WidgetStatDComponent,
        WidgetStatEComponent,
        WidgetStatFComponent], imports: [CommonModule,
        CardModule,
        GridModule], exports: [WidgetStatAComponent,
        WidgetStatBComponent,
        WidgetStatCComponent,
        WidgetStatDComponent,
        WidgetStatEComponent,
        WidgetStatFComponent] });
WidgetModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: WidgetModule, imports: [CommonModule,
        CardModule,
        GridModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: WidgetModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        WidgetStatAComponent,
                        WidgetStatBComponent,
                        WidgetStatCComponent,
                        WidgetStatDComponent,
                        WidgetStatEComponent,
                        WidgetStatFComponent
                    ],
                    imports: [
                        CommonModule,
                        CardModule,
                        GridModule,
                    ],
                    exports: [
                        WidgetStatAComponent,
                        WidgetStatBComponent,
                        WidgetStatCComponent,
                        WidgetStatDComponent,
                        WidgetStatEComponent,
                        WidgetStatFComponent
                    ]
                }]
        }] });
