// @coreui/angular/esm2020/lib/popover/popover.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { PopoverDirective } from './popover.directive';
import { PopoverComponent } from './popover/popover.component';
import * as i0 from "@angular/core";
export class PopoverModule {
}
PopoverModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: PopoverModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
PopoverModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: PopoverModule, declarations: [PopoverDirective,
        PopoverComponent], imports: [CommonModule], exports: [PopoverDirective,
        PopoverComponent] });
PopoverModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: PopoverModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: PopoverModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [
                        PopoverDirective,
                        PopoverComponent
                    ],
                    exports: [
                        PopoverDirective,
                        PopoverComponent
                    ],
                    imports: [
                        CommonModule
                    ]
                }]
        }] });
