// @coreui/angular/esm2020/lib/image/img.module.mjs
import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { ImgDirective } from './img.directive';
import * as i0 from "@angular/core";
export class ImgModule {
}
ImgModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: ImgModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
ImgModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "15.2.1", ngImport: i0, type: ImgModule, declarations: [ImgDirective], imports: [CommonModule], exports: [ImgDirective] });
ImgModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: ImgModule, imports: [CommonModule] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "15.2.1", ngImport: i0, type: ImgModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [
                        CommonModule
                    ],
                    exports: [
                        ImgDirective
                    ],
                    declarations: [
                        ImgDirective
                    ],
                }]
        }] });
